### PR TITLE
Convert e2e test script logic into python code

### PR DESCRIPTION
**Pull request by bparees**
also refactor some of the e2e helper code into utils modules

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by bparees** [comment]
@onmete  @nemerna this PR will be relevant to your fine grained permissions testing as it introduces what i hope you will find are helpful ways to manage additional users/permissions on the cluster.

File: tests/e2e/cluster_utils.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+"""Utilities for interacting with the OpenShift cluster."""
+
+import subprocess
+
+
+def create_user(name: str):
+    """Create a service account user for testing."""
+    try:
+        subprocess.run(
```
Comment (tisnik) [inline tests/e2e/cluster_utils.py:9]: prob. refactor this repeated command (all the keyword args are the same)

**Comment by bparees** [comment]
/assign @tisnik 

there's more to be done here, but this should encourage a bit more structure to our e2e suite, and starts getting rid of some of the bash scripting.

File: tests/e2e/cluster_utils.py (hunk 1 above)
Comment (bparees) [reply tests/e2e/cluster_utils.py]: refactor pushed in commit `pavel1`, please see if that's what you meant

**Review by tisnik — COMMENTED** [review]
it looks good

**Pull request merged**